gen/nvml/lib.go

```diff
@@ -255,3 +268,36 @@ func (l *library) updateVersionedSymbols() {
 		nvmlVgpuInstanceGetLicenseInfo = nvmlVgpuInstanceGetLicenseInfo_v2
 	}
 }
+
+// LibraryOption represents a functional option to configure the underlying NVML library
+type LibraryOption func(*library)
+
+// WithLibraryPath provides an option to set the library name to be used by the NVML library.
+func WithLibraryPath(path string) LibraryOption {
+	return func(l *library) {
+		l.path = path
+	}
+}
+
+// SetLibraryOptions applies the specified options to the NVML library.
+// If this is called when a library is already loaded, and error is raised.
+func SetLibraryOptions(opts ...LibraryOption) error {
```

Comment:
I think it was in the last MR, but can you explain the purpose of the `Default()` method? I'm just thinking about it from a callsite perspective --- `nvml.Default()` doesn't really imply to me that I'm getting a reference to the default implementation of the nvml Interface. When / how do you actually expect this to be used?

Reply:
I see -- its specifically so we can get at that Library method -- can we just make a top level Library() function then?

Reply:
Would `GetLibrary()` be ok? We would either need to rename this or the `Library` interface.

Reply:
Hmm, right -- I'd rather the top level call be the same as what we would eventually call on an instance of `Interface`. Maybe that becomes `GetLibrary()` as well?

Reply:
OK. I renamed the function `GetLibrary` and also included an implementation at the package level.